azure/services/disks/client.go

```diff
@@ -21,48 +21,81 @@ import (
 
 	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2021-04-01/compute"
 	"github.com/Azure/go-autorest/autorest"
+	azureautorest "github.com/Azure/go-autorest/autorest/azure"
+	"github.com/pkg/errors"
+
 	"sigs.k8s.io/cluster-api-provider-azure/azure"
+	"sigs.k8s.io/cluster-api-provider-azure/util/reconciler"
 	"sigs.k8s.io/cluster-api-provider-azure/util/tele"
 )
 
 // Client wraps go-sdk.
-type client interface {
-	Delete(context.Context, string, string) error
+type Client interface {
+	DeleteAsync(context.Context, azure.ResourceSpecGetter) (azureautorest.FutureAPI, error)
+	Result(context.Context, azureautorest.FutureAPI, string) (interface{}, error)
+	IsDone(context.Context, azureautorest.FutureAPI) (bool, error)
 }
 
 // AzureClient contains the Azure go-sdk Client.
-type azureClient struct {
+type AzureClient struct {
```

Comment:
why change azureClient to public? only the clients that are being used in other packages should be public and it doesn't look like we are changing that requirement in this PR

ref https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1013